GoogleForCreators/web-stories-wp
# Shopping: Allow sorting products

## Context

Add support to sort products by date, title, and price in ascending or descending order.



## Summary

- Adds sort by dropdown to the product sidebar and updates the search queries to allow sorting.



## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions

- Keyboard handling for dropdown options doesn't seem to be working.






This PR can be tested by following these steps:

1. Enable shopping + a shopping vendor
2. Visit the shopping tab
3. Select a sort order from `sort` dropdown

* Note: the dropdown doesn't match the designs (using the default dropdown)
<img width="324" alt="Screen Shot 2022-05-24 at 5 18 34 AM" src="https://user-images.githubusercontent.com/62242/169997218-bafba201-a267-409e-a571-bae4474fac25.png">

4. Ensure the products are sorted as per the sort order.



## Reviews

### Does this PR have a security-related impact?

- New query params orderby & order



### Does this PR change what data or activity we track or use?

No



### Does this PR have a legal-related impact?

No



## Checklist




---



Fixes #11253
